test/integration/tcp_tunneling_integration_test.cc

```diff
@@ -173,6 +173,35 @@ TEST_P(ConnectTerminationIntegrationTest, BuggyHeaders) {
   ASSERT_FALSE(response_->reset());
 }
 
+TEST_P(ConnectTerminationIntegrationTest, BasicMaxStreamDuration) {
+  config_helper_.addConfigModifier([](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+    auto* static_resources = bootstrap.mutable_static_resources();
+    auto* cluster = static_resources->mutable_clusters(0);
+    auto* http_protocol_options = cluster->mutable_common_http_protocol_options();
+    http_protocol_options->mutable_max_stream_duration()->MergeFrom(
+        ProtobufUtil::TimeUtil::MillisecondsToDuration(1000));
+  });
+
+  initialize();
+  fake_upstreams_[0]->set_allow_unexpected_disconnects(true);
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+
+  auto encoder_decoder = codec_client_->startRequest(connect_headers_);
+  request_encoder_ = &encoder_decoder.first;
+  auto response = std::move(encoder_decoder.second);
+
+  ASSERT_TRUE(fake_upstreams_[0]->waitForHttpConnection(*dispatcher_, fake_upstream_connection_));
```

Comment:
Can you actually pass some data here to make this test a bit more realistic? I think you want to wait for a raw TCP connection in this case and pass a bit of data back and forth before waiting?

Reply:
Reworked. My expecting behavior is only that waiting for upstream TCP connection and causing disconnection of TCP by this timer. Does my recognition correspond with yours?

Reply:
Can you modify the test (or add a new one) to actually pass some data back and forth and then wait for the timeout? That will make for a better test. Thank you!

/wait

Reply:
like this? followed other connect termination testing.